python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1705,6 +1705,60 @@ def get_materialization_count_by_partition(
 
         return materialization_count_by_partition
 
+    def get_latest_asset_materialization_planned_event_by_partition(
+        self, asset_key: AssetKey
+    ) -> Mapping[str, EventLogRecord]:
+        check.inst_param(asset_key, "asset_key", AssetKey)
+
+        latest_event_subquery = (
```

Comment:
nit: probs should rename this variable to something like `partitioned_materialization_planned_events` or something like that